src/Device/Driver/AirControlDisplay.cpp

```diff
@@ -201,6 +202,19 @@ ACDDevice::OnSensorUpdate(const MoreData &basic)
   }
 }
 
+void
+ACDDevice::OnCalculatedUpdate(const MoreData &basic,[[maybe_unused]] const DerivedInfo &calculated)
+{
+   NullOperationEnvironment env;
+
+   if (basic.settings.qnh_available.IsValid()){
+	 char buffer[100];
+     unsigned qnh = basic.settings.qnh.GetPascal();
+     sprintf(buffer,"PAAVC,S,ALT,QNH,%u",qnh);
+     PortWriteNMEA(port, buffer, env);
```

Comment:
This function can block. It must not be called from within `OnCalculatedUpdate()`, because the latter must never block.

Reply:
you mean this function: sprintf?

Reply:
No, 
1. this isn't the line I commented on
2. how is sprintf supposed to block?

PortWriteNMEA() can block, and you circumvented the API protection by using a NullOperationEnvironment, just to make it compile, but without knowing the implications. Don't do random stuff just to make it compile! If you don't understand your own code, it's bad code.